web_monitoring/diffing_server.py

```diff
@@ -78,12 +79,33 @@ def __init__(self, url):
 
 class MockResponse:
     "An HTTPResponse-like object for local file:/// requests."
-    def __init__(self, url, body, headers):
+    def __init__(self, url, body, headers=None):
         self.request = MockRequest(url)
         self.body = body
         self.headers = headers
         self.error = None
 
+        if self.headers is None:
+            self.headers = {}
```

Comment:
This check could probably be simplified by just updating the setter for headers above on line 85:

```py
self.headers = headers or {}
```

(It’s not strictly the same, but probably good enough for this case.)